File: Components/WinForms/CS/SchedulerMainDemo/Modules/Restrictions.cs
using System;

namespace DevExpress.XtraScheduler.Demos {
	public partial class RestrictionsModule : DevExpress.XtraScheduler.Demos.TutorialControl {

		public RestrictionsModule() {
			// This call is required by the Windows Form Designer.
			InitializeComponent();

			// TODO: Add any initialization after the InitializeComponent call
		}
		public override SchedulerControl PrintingSchedulerControl { get { return schedulerControl; } }

		private void RestrictionsModule_Load(object sender, System.EventArgs e) {
            CarsDataHelper.FillData(schedulerControl, schedulerStorage);
            UpdateControls();
		}
		private void UpdateControls() {
			SchedulerOptionsCustomization opts = schedulerControl.OptionsCustomization;
			chkAllowConflicts.Checked = opts.AllowAppointmentConflicts == AppointmentConflictsMode.Allowed;
			chkAllowCopy.Checked = opts.AllowAppointmentCopy == UsedAppointmentType.All;
			chkAllowCreate.Checked = opts.AllowAppointmentCreate == UsedAppointmentType.All;
			chkAllowDelete.Checked = opts.AllowAppointmentDelete == UsedAppointmentType.All;
			chkAllowDrag.Checked = opts.AllowAppointmentDrag == UsedAppointmentType.All;
			chkAllowDragBetweenResources.Checked = opts.AllowAppointmentDragBetweenResources == UsedAppointmentType.All;
			chkAllowEdit.Checked = opts.AllowAppointmentEdit == UsedAppointmentType.All;
			chkAllowInplaceEditor.Checked = opts.AllowInplaceEditor == UsedAppointmentType.All;
			chkAllowMultiSelect.Checked = opts.AllowAppointmentMultiSelect;
			chkAllowResize.Checked = opts.AllowAppointmentResize == UsedAppointmentType.All;
		}
		private void chkAllowCreate_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentCreate = chkAllowCreate.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}

		private void chkAllowEdit_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentEdit = chkAllowEdit.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}

		private void chkAllowResize_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentResize = chkAllowResize.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}

		private void chkAllowConflicts_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentConflicts = chkAllowConflicts.Checked ? AppointmentConflictsMode.Allowed : AppointmentConflictsMode.Forbidden;
		}

		private void chkAllowDragBetweenResources_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentDragBetweenResources = chkAllowDragBetweenResources.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}

		private void chkAllowDelete_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentDelete = chkAllowDelete.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}

		private void chkAllowCopy_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentCopy = chkAllowCopy.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}

		private void chkAllowDrag_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentDrag = chkAllowDrag.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}

		private void chkAllowMultiSelect_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowAppointmentMultiSelect = chkAllowMultiSelect.Checked;
		}

		private void chkAllowInplaceEditor_CheckedChanged(object sender, System.EventArgs e) {
			schedulerControl.OptionsCustomization.AllowInplaceEditor = chkAllowInplaceEditor.Checked ? UsedAppointmentType.All : UsedAppointmentType.None;
		}
	}
}

File: Components/WinForms/VB/SchedulerMainDemo/Modules/Restrictions.vb
Imports Microsoft.VisualBasic
Imports System

Namespace DevExpress.XtraScheduler.Demos
	Partial Public Class RestrictionsModule
		Inherits DevExpress.XtraScheduler.Demos.TutorialControl

		Public Sub New()
			' This call is required by the Windows Form Designer.
			InitializeComponent()

			' TODO: Add any initialization after the InitializeComponent call
		End Sub
		Public Overrides ReadOnly Property PrintingSchedulerControl() As SchedulerControl
			Get
				Return schedulerControl
			End Get
		End Property

		Private Sub RestrictionsModule_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
			CarsDataHelper.FillData(schedulerControl, schedulerStorage)
			UpdateControls()
		End Sub
		Private Sub UpdateControls()
			Dim opts As SchedulerOptionsCustomization = schedulerControl.OptionsCustomization
			chkAllowConflicts.Checked = opts.AllowAppointmentConflicts = AppointmentConflictsMode.Allowed
			chkAllowCopy.Checked = opts.AllowAppointmentCopy = UsedAppointmentType.All
			chkAllowCreate.Checked = opts.AllowAppointmentCreate = UsedAppointmentType.All
			chkAllowDelete.Checked = opts.AllowAppointmentDelete = UsedAppointmentType.All
			chkAllowDrag.Checked = opts.AllowAppointmentDrag = UsedAppointmentType.All
			chkAllowDragBetweenResources.Checked = opts.AllowAppointmentDragBetweenResources = UsedAppointmentType.All
			chkAllowEdit.Checked = opts.AllowAppointmentEdit = UsedAppointmentType.All
			chkAllowInplaceEditor.Checked = opts.AllowInplaceEditor = UsedAppointmentType.All
			chkAllowMultiSelect.Checked = opts.AllowAppointmentMultiSelect
			chkAllowResize.Checked = opts.AllowAppointmentResize = UsedAppointmentType.All
		End Sub
		Private Sub chkAllowCreate_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowCreate.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentCreate = If(chkAllowCreate.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub

		Private Sub chkAllowEdit_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowEdit.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentEdit = If(chkAllowEdit.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub

		Private Sub chkAllowResize_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowResize.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentResize = If(chkAllowResize.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub

		Private Sub chkAllowConflicts_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowConflicts.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentConflicts = If(chkAllowConflicts.Checked, AppointmentConflictsMode.Allowed, AppointmentConflictsMode.Forbidden)
		End Sub

		Private Sub chkAllowDragBetweenResources_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowDragBetweenResources.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentDragBetweenResources = If(chkAllowDragBetweenResources.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub

		Private Sub chkAllowDelete_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowDelete.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentDelete = If(chkAllowDelete.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub

		Private Sub chkAllowCopy_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowCopy.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentCopy = If(chkAllowCopy.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub

		Private Sub chkAllowDrag_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowDrag.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentDrag = If(chkAllowDrag.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub

		Private Sub chkAllowMultiSelect_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowMultiSelect.CheckedChanged
			schedulerControl.OptionsCustomization.AllowAppointmentMultiSelect = chkAllowMultiSelect.Checked
		End Sub

		Private Sub chkAllowInplaceEditor_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles chkAllowInplaceEditor.CheckedChanged
			schedulerControl.OptionsCustomization.AllowInplaceEditor = If(chkAllowInplaceEditor.Checked, UsedAppointmentType.All, UsedAppointmentType.None)
		End Sub
	End Class
End Namespace
